GCC warning: unused-but-set-variable warning

When compiling zcbor for my target hardware, GCC is giving me a `unused-but-set-variable` warning.

```
C:/Users/<path>/zcbor/src/zcbor_decode.c: In function 'zcbor_any_skip':
C:/Users/<path>/zcbor/src/zcbor_decode.c:1461:17: error: variable 'additional' set but not used [-Werror=unused-but-set-variable]
 1461 |         uint8_t additional = ZCBOR_ADDITIONAL(*state->payload);
      |                 ^~~~~~~~~~
cc1.exe: all warnings being treated as errors
```

This variable doesn't seem to be used. I can easily comment out the usage of the variable or disable the warning so it is not of any real concern, I just thought I'd raise the issue to be thorough.

--- a/src/zcbor_decode.c
+++ b/src/zcbor_decode.c
@@ -1458,7 +1458,6 @@ bool zcbor_any_skip(zcbor_state_t *state, void *result)
 
 	INITIAL_CHECKS();
 	zcbor_major_type_t major_type = ZCBOR_MAJOR_TYPE(*state->payload);
-	uint8_t additional = ZCBOR_ADDITIONAL(*state->payload);
 	uint64_t value = 0; /* In case of indefinite_length_array. */
 	zcbor_state_t state_copy;
 
@@ -1472,7 +1471,6 @@ bool zcbor_any_skip(zcbor_state_t *state, void *result)
 		}
 		ZCBOR_ERR_IF(state_copy.payload >= state_copy.payload_end, ZCBOR_ERR_NO_PAYLOAD);
 		major_type = ZCBOR_MAJOR_TYPE(*state_copy.payload);
-		additional = ZCBOR_ADDITIONAL(*state_copy.payload);
 	}
 
 	bool indefinite_length_array = false;

